lib/collection/property-list.js

```diff
@@ -289,6 +316,37 @@ _.assign(PropertyList.prototype, /** @lends PropertyList.prototype */ {
     },
 
     /**
+     * Add or update values from a source list.
+     *
+     * @param {PropertyList|Array} source
+     * @param {Boolean=false} prune Setting this to `true` will cause the extra items from the list to be deleted
+     */
+    assimilate: function (source, prune) {
+        var members = PropertyList.isPropertyList(source) ? source.members : source,
+            list = this,
+            indexer = list._postman_listIndexKey,
+            sourceKeys = {}; // keeps track of added / updated keys for later exclusion
+
+        if (!_.isArray(members)) {
+            return;
+        }
+
+        members.forEach(function (item) {
+            if (!(item && item.hasOwnProperty(indexer))) { return; }
+            list.upsert(item);
+            sourceKeys[item[indexer]] = true;
+        });
+
+        // now remove any variable that is not in source object
+        // @note - using direct `this.reference` list of keys here so that we can mutate the list while iterating
+        // on it
+        prune && _.forEach(_.keys(list.reference), function (key) {
```

Comment:
`_.forEach` works well with objects as well, the extra `_.keys` call can be avoided.